components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -499,11 +512,17 @@ bool ViewCounterService::CanShowSponsoredImages() const {
 
   if (images_data->grace_period &&
       local_state_->FindPreference(metrics::prefs::kInstallDate)) {
-    const base::Time installation_date = base::Time::FromSecondsSinceUnixEpoch(
+    const base::Time install_date = base::Time::FromSecondsSinceUnixEpoch(
         local_state_->GetInt64(metrics::prefs::kInstallDate));
-    if (base::Time::Now() < installation_date + *images_data->grace_period) {
+    const base::Time grace_period_ends_at =
+        install_date + *images_data->grace_period;
+
+    if (base::Time::Now() < grace_period_ends_at) {
       // We don't show sponsored images if the user has installed Brave within
       // the grace period.
+      VLOG(1) << "Sponsored images not shown: Grace period after "
+                 "installation is still active until "
+              << grace_period_ends_at;
```

Comment:
Added to help QA test a fresh install, otherwise it would look to the user like no ads are being served. Requires a logging verbosity level of 1 or higher.